index.bs

```diff
@@ -0,0 +1,225 @@
+                1. Let <var>horizontalDistance</var> be abs(<a>previous frame starting point</a>'s x-value - <a>current frame starting point</a>'s x-value) / <var>viewportDistance</var>.
+                1. Take the max(<var>verticalDistance</var>, <var>horizontalDistance</var>).
+        1. If <var>unstable</var> is true, add a tuple to <var>unstableElements</var> with:
+            1. <a>current visual representation</a> of <var>element</var>
+            1. <a>previous visual representation</a> of <var>element</var>
+            1. <var>shiftFraction</var> of <var>element</var>:
+        1. Set <var>element</var>'s <a>previous frame starting point</a> point to <a>current frame starting point</a>.
+        1. Set <var>element</var>'s <a>previous visual representation</a> to the <a>current visual representation</a>.
+    1. Call <a>compute the layout shift</a> with <var>unstableElements</var>.
+</div>
+
+Compute the layout shift {#sec-compute-layout-shift}
+-----------------------------------------------------
+
+<div algorithm="compute the layout shift">
+    When asked to to <dfn export>compute the layout shift</dfn>, with <var>unstableElements</var>, a list of impacted <a>element</a>'s on the page, as input, run the following steps:
+
+    1. Let <var>impactedRegion</var> be a two-dimensional geometric region which is initially empty (that is, it contains no points).
+    1. Let <var>maxShiftFraction</var> be initially set to 0.
+    1. For each tuple <var>tuple</var> in <var>unstableElements</var>:
+        1. If <a>previous visual representation</a> is not null, expand <var>impactedRegion</var> to additionally contain all points that lie within <var>tuple</var>'s <a>current visual representation</a>, <var>tuple</var>'s <a>previous visual representation</a>, or both.
+        1. Set <var>maxShiftFraction</var> to <var>shiftFraction</var> if <var>tuple</var>’s <var>shiftFraction</var> > <var>maxShiftFraction</var>.
+    1. Let <var>impactedFraction</var> be the area of the <a>initial containing block</a> of the <var>impactedRegion</var> divided by the area of the current <a>viewport</a>.
+    1. Set <var>layoutShift</var> to the <var>impactedFraction</var> multiplied by the <var>maxShiftFraction</var>.  
+    1. If <var>layoutShift</var> is greater than 0, create a new {{LayoutShift}} object |newEntry| and set its attributes as follows:
+        1. Set |newEntry|'s {{PerformanceEntry/name}} attribute to <code>"layout-shift"</code>.
+        1. Set |newEntry|'s {{PerformanceEntry/entryType}} attribute to <code>"layoutShift"</code>.
+        1. Set |newEntry|'s {{PerformanceEntry/startTime}} attribute to <a>current high resolution time</a>.
+        1. Set |newEntry|'s {{PerformanceEntry/duration}} attribute to 0.
+        1. Set |newEntry|'s {{LayoutShift/value}} attribute to <var>layoutShift</var>.
+    1. <a href="https://w3c.github.io/performance-timeline/#dfn-queue-a-performanceentry">Queue the PerformanceEntry</a> |newEntry| object.
```

Comment:
This should also be a substep of the step asking layoutShift > 0.

Reply:
Done.